File: arc3-api/src/v1/db/Appeal.js
const mongoose = require('mongoose');

const AppealSchema = new mongoose.Schema({
    _id: String,
    userSnowflake: String,
    bannedBy: String,
    action: String,
    appealContent: String,
    nextappeal: String
});

const Appeal = mongoose.model('Appeal', AppealSchema);

module.exports = Appeal;

File: Arc3/Core/Schema/Appeal.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace Arc3.Core.Schema;

public class Appeal {

  [BsonId]
  [BsonRepresentation(BsonType.String)]
  public string Id { get; set; } = String.Empty;

  [BsonElement("userSnowflake")]
  public long UserSnowflake { get; set; }

  [BsonElement("nextappeal")]
  public long NextAppeal { get; set; }
  
  [BsonElement("bannedBy")]
  public string BannedBy { get; set; }
  
  [BsonElement("appealContent")]
  public string AppealContent { get; set; }

  [BsonElement("action")]
  public string Action { get; set; }

}
